pkg/duplicateevents/duplicated_event_patterns.go

```diff
@@ -361,6 +368,24 @@ func isOauthApiserverProbeErrorConnectionRefusedFailed(monitorEvent monitorapi.E
 	return IsOperatorMatchRegexMessage(monitorEvent, "openshift-oauth-apiserver", regExp), nil
 }
 
+// reason/ErrorReconcilingNode roles/worker [k8s.ovn.org/node-chassis-id annotation not found for node ci-op-nzi4gt1b-3efb3-ggmhb-worker-centralus2-jzx86, macAddress annotation not found for node "ci-op-nzi4gt1b-3efb3-ggmhb-worker-centralus2-jzx86" , k8s.ovn.org/l3-gateway-config annotation not found for node "ci-op-nzi4gt1b-3efb3-ggmhb-worker-centralus2-jzx86"]
+func isErrorReconcilingNode(monitorEvent monitorapi.EventInterval, _ *rest.Config, count int) (bool, error) {
+	regExp := regexp.MustCompile(ErrorReconcilingNode)
+	return regExp.MatchString(monitorEvent.String()) && count < DuplicateSingleNodeEventThreshold, nil
+}
+
+// reason/FailedScheduling 0/6 nodes are available: 2 node(s) didn't match Pod's node affinity/selector, 2 node(s) didn't match pod anti-affinity rules, 2 node(s) were unschedulable. preemption: 0/6 nodes are available: 2 node(s) didn't match pod anti-affinity rules, 4 Preemption is not helpful for scheduling..
+func isFailedScheduling(monitorEvent monitorapi.EventInterval, _ *rest.Config, _ int) (bool, error) {
+	regExp := regexp.MustCompile(FailedScheduling)
+	return IsOperatorMatchRegexMessage(monitorEvent, "openshift-route-controller-manager", regExp), nil
+}
+
+// reason/OperatorStatusChanged Status for clusteroperator/csi-snapshot-controller changed: Progressing changed from False to True ("CSISnapshotControllerProgressing: Waiting for Deployment to deploy pods")
+func isOperatorStatusChanged(monitorEvent monitorapi.EventInterval, _ *rest.Config, _ int) (bool, error) {
+	regExp := regexp.MustCompile(OperatorStatusChanged)
+	return IsOperatorMatchRegexMessage(monitorEvent, "openshift-kube-apiserver-operator", regExp), nil
+}
+
```

Comment:
Let's add unit tests for these 3 new functions that validate true / false conditions as expected.